src/Nethermind/Nethermind.Blockchain/Receipts/PersistentReceiptStorage.cs

```diff
@@ -241,7 +202,13 @@ public void Insert(Block block, params TxReceipt[] txReceipts)
             }
             
             _receiptsCache.Set(block.Hash, txReceipts);
-            ReceiptsInserted?.Invoke(this, new ReceiptsEventArgs(block.Header, txReceipts));
+
+            // Subscription with new receipts is invoked by NewHeadBlock event.
+            // Here we want to invoke ReceiptsInserted event only to send removed receipts (after reorg).
```

Comment:
I don't really like that, this should be independent component, here we are mixing abstraction layers.
We might want to change it so ReceiptsEventArgs has additional info if the receipts were deleted or if the receipts were fully or partially recovered and if they were inserted for the first or second time and then amend the users of this event to handle correctly?

Reply:
This event was added specifically for LogsSubscription, but later subscription trigger for new logs was moved to NewHeadBlock event and now ReceiptsInserted event is useful only for removed ones: https://github.com/NethermindEth/nethermind/blob/master/src/Nethermind/Nethermind.JsonRpc/Modules/Subscribe/LogsSubscription.cs#L82-L89
I wanted to reduce number of event invokes, but I don't like that it mixes abstraction layers too. I can remove 'if' statement and it will still work properly - it's filtered in LogsSubscription anyway. Other option and imo better is to change the name to ReceiptsRemoved and then it will be logically consistent and there will be no unnecessary invokes. What do you prefer?